pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -245,9 +313,137 @@ func (r *ReconcileNodeMaintenance) stopNodeMaintenance(nodeName string) error {
 		return err
 	}
 
+	if r.isLeaseSupported {
+		reqLogger.Info("Lease object supported, deleting lease")
+		if err := r.deleteLease(nodeName); err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
+func (r *ReconcileNodeMaintenance) deleteLease(nodeName string) error {
+
+		nName := apitypes.NamespacedName{Namespace: LeaseNamespace, Name: nodeName}
+		lease := &coordv1beta1.Lease{}
+
+		if err := r.client.Get(context.TODO(), nName, lease); err != nil {
+			return err
+		}
+		r.client.Delete(context.TODO(), lease);
+		return nil
+}
+
+func (r *ReconcileNodeMaintenance) obtainLease(node *corev1.Node, durationInSeconds int, reqLogger *log.Entry) (bool, reconcile.Result) {
+
+	lease, needUpdate, err := r.createLease(node, durationInSeconds)
+
+	if err != nil {
+		reqLogger.Debugf("MaintOn: failed to create lease error=%v", err)
+		return false, reconcile.Result{Requeue: true, RequeueAfter: RequeWaitTime}
+	}
+
+	if needUpdate == false {
+		return true, reconcile.Result{}
+	}
+
+	reqLogger.Info("MaintOn: update lease - lease duration change requested")
+
+	if _, err = r.updateLease(node, lease, durationInSeconds, TransitionNoChange, false, reqLogger); err != nil {
+		return false, reconcile.Result{Requeue: true, RequeueAfter: RequeWaitTime}
+	}
+
+	return true, reconcile.Result{}
+}
+
+func (r *ReconcileNodeMaintenance) createLease(node *corev1.Node, durationInSeconds int) (*coordv1beta1.Lease, bool, error) {
+	nodeName := node.ObjectMeta.Name
+
+	holderIdentity := LeaseHolderIdentity
+
+	owner := &metav1.OwnerReference{
+		APIVersion: "v1",
+		Kind:       "Node",
+		Name:       nodeName,
+		UID:        node.ObjectMeta.UID,
+	}
+	leaseTransitions := int32(1)
+	microTimeNow := metav1.NewMicroTime(time.Now())
+
+	tmpDurationInSeconds := int32(durationInSeconds)
+	var refDuration *int32 =  &tmpDurationInSeconds
+
+	lease := &coordv1beta1.Lease{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            nodeName,
+			Namespace:       LeaseNamespace,
+			OwnerReferences: []metav1.OwnerReference{*owner},
+		},
+		Spec: coordv1beta1.LeaseSpec{
+			HolderIdentity:       &holderIdentity,
+			LeaseDurationSeconds: refDuration,
+			AcquireTime:          &microTimeNow,
+			RenewTime:            &microTimeNow,
+			LeaseTransitions:     &leaseTransitions,
+		},
+	}
+
+	if err := r.client.Create(context.TODO(), lease); err != nil {
+		if errors.IsAlreadyExists(err) {
+			return nil, true, nil
+		}
+		return nil, false, err
+	}
+	return lease, false,  nil
+}
+
+func (r *ReconcileNodeMaintenance) updateLease(node *corev1.Node, lease *coordv1beta1.Lease, durationInSeconds int, transitions TransitionAction, resetDuration bool, reqLogger *log.Entry) (*coordv1beta1.Lease, error) {
```

Comment:
is there a way perhaps we don't need to pass reqLogger here ?

Reply:
I made the same comment 2 weeks ago :-(
https://github.com/kubevirt/node-maintenance-operator/pull/64/files#r394761300

Reply:
done